avocado/utils/network/hosts.py

```diff
@@ -48,7 +48,7 @@ class Host:
     """
 
     def __init__(self, host):
-        if isinstance(self, Host):
+        if type(self) == Host:
```

Comment:
Btw, there is another approach to avoid performing this check by invoking `type` directly, that is we could make the `Host` be an abstract class with its `__init__` method being an abstract method.

```
$ cat /tmp/t.py 
from abc import ABCMeta, abstractmethod


class Foo(metaclass=ABCMeta):

    @abstractmethod
    def __init__(self, host):
        self.host = host


class Bar(Foo):

    def __init__(self, host='localhost'):
        super().__init__(host)


b = Bar()
print(b.host)
f = Foo('localhost')


$ python3 /tmp/t.py 
localhost
Traceback (most recent call last):
  File "/tmp/t.py", line 19, in <module>
    f = Foo('localhost')
        ^^^^^^^^^^^^^^^^
TypeError: Can't instantiate abstract class Foo with abstract method __init__
```